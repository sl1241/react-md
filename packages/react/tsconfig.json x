{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types",
    "composite": true,
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "jsx": "preserve",
    "jsxImportSource": "voby",
    "types": [
      "voby/dist/jsx/jsx"
    ],
  },
  "exclude": [
    "vite.config.ts",
"vite.config.umd.ts",
    "types",
  ],
}
