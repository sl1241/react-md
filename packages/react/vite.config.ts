import { defineConfig } from 'vite'

const config = defineConfig({
    build: {
        minify: false,
        lib: {
            entry: ["./src/index.ts"],
            name: "react",
            formats: ['cjs', 'es', 'umd'],
            fileName: (format: string, entryName: string) => `${entryName}.${format}.js`
        },
        // rollupOptions: {
        //   output: {
        //     preserveModules: true,
        //   }
        // },
    },


    esbuild: {
        jsx: 'automatic',
    },

    css: {
        modules: {
            scopeBehaviour: 'global',
            localsConvention: 'camelCase',
        }
    },
})



export default config
