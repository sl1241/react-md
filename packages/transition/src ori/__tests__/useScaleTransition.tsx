import type { ReactElement } from "react";
import { useState } from "react";
import { act, fireEvent, render } from "@testing-library/react";

import type { ScaleTransitionHookOptions } from "../useScaleTransition";
import { useScaleTransition } from "../useScaleTransition";

interface TestProps
  extends Omit<ScaleTransitionHookOptions<HTMLElement>, "transitionIn"> {
  defaultTransitionIn?: boolean;
}

function Test({
  defaultTransitionIn = false,
  ...options
}: TestProps): ReactElement {
  const [transitionIn, setTransitionIn] = useState(defaultTransitionIn);
  const { elementProps, rendered } = useScaleTransition({
    transitionIn,
    ...options,
  });
  return (
    <>
      <button type="button" onClick={() => setTransitionIn((p) => !p)}>
        Toggle
      </button>
      {rendered && (
        <div {...elementProps} data-testid="element">
          This is some content.
        </div>
      )}
    </>
  );
}

describe("ScaleTransition", () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllTimers();
  });

  it("should default using the horizontal transition", () => {
    const { container, getByRole, getByTestId } = render(<Test />);
    const getElement = (): HTMLElement => getByTestId("element");
    const toggle = getByRole("button");

    expect(getElement).toThrow();
    expect(container).toMatchSnapshot();

    fireEvent.click(toggle);
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    fireEvent.click(toggle);
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).toThrow();
    expect(container).toMatchSnapshot();
  });

  it("should render correctly for a vertical transition", () => {
    const { container, getByRole, getByTestId } = render(<Test vertical />);
    const getElement = (): HTMLElement => getByTestId("element");
    const toggle = getByRole("button");

    expect(getElement).toThrow();
    expect(container).toMatchSnapshot();

    fireEvent.click(toggle);
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    fireEvent.click(toggle);
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).toThrow();
    expect(container).toMatchSnapshot();
  });

  it("should merge the className and allow to be rendered by default", () => {
    const { container, getByRole, getByTestId } = render(
      <Test className="custom-class" defaultTransitionIn />
    );

    const getElement = (): HTMLElement => getByTestId("element");
    const toggle = getByRole("button");

    expect(getElement()).toHaveClass("custom-class");
    expect(container).toMatchSnapshot();

    fireEvent.click(toggle);
    expect(getElement()).toHaveClass("custom-class");
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).toThrow();
    expect(container).toMatchSnapshot();

    fireEvent.click(toggle);
    expect(getElement()).toHaveClass("custom-class");
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement()).toHaveClass("custom-class");
    expect(container).toMatchSnapshot();
  });

  it("should correctly call the transition callbacks", () => {
    const onEnter = jest.fn();
    const onEntering = jest.fn();
    const onEntered = jest.fn();
    const onExit = jest.fn();
    const onExiting = jest.fn();
    const onExited = jest.fn();
    const props: TestProps = {
      onEnter,
      onEntering,
      onEntered,
      onExit,
      onExiting,
      onExited,
    };

    const { getByRole } = render(<Test {...props} />);
    const toggle = getByRole("button");

    expect(onEnter).not.toBeCalled();
    expect(onEntering).not.toBeCalled();
    expect(onEntered).not.toBeCalled();
    expect(onExit).not.toBeCalled();
    expect(onExiting).not.toBeCalled();
    expect(onExited).not.toBeCalled();

    fireEvent.click(toggle);
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).not.toBeCalled();
    expect(onExit).not.toBeCalled();
    expect(onExiting).not.toBeCalled();
    expect(onExited).not.toBeCalled();

    act(() => {
      jest.runAllTimers();
    });
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).toBeCalledTimes(1);
    expect(onExit).not.toBeCalled();
    expect(onExiting).not.toBeCalled();
    expect(onExited).not.toBeCalled();

    fireEvent.click(toggle);
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).toBeCalledTimes(1);
    expect(onExit).toBeCalledTimes(1);
    expect(onExiting).toBeCalledTimes(1);
    expect(onExited).not.toBeCalled();

    act(() => {
      jest.runAllTimers();
    });
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).toBeCalledTimes(1);
    expect(onExit).toBeCalledTimes(1);
    expect(onExiting).toBeCalledTimes(1);
    expect(onExited).toBeCalledTimes(1);
  });
});
