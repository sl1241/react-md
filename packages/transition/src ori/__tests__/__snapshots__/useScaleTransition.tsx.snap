// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScaleTransition should default using the horizontal transition 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;

exports[`ScaleTransition should default using the horizontal transition 2`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="rmd-transition--scale-enter rmd-transition--scale-enter-active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should default using the horizontal transition 3`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should default using the horizontal transition 4`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="rmd-transition--scale-exit rmd-transition--scale-exit-active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should default using the horizontal transition 5`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;

exports[`ScaleTransition should merge the className and allow to be rendered by default 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="custom-class"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should merge the className and allow to be rendered by default 2`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="custom-class rmd-transition--scale-exit rmd-transition--scale-exit-active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should merge the className and allow to be rendered by default 3`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;

exports[`ScaleTransition should merge the className and allow to be rendered by default 4`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="custom-class rmd-transition--scale-enter rmd-transition--scale-enter-active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should merge the className and allow to be rendered by default 5`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="custom-class"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should render correctly for a vertical transition 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;

exports[`ScaleTransition should render correctly for a vertical transition 2`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="rmd-transition--scale-y-enter rmd-transition--scale-y-enter-active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should render correctly for a vertical transition 3`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should render correctly for a vertical transition 4`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="rmd-transition--scale-y-exit rmd-transition--scale-y-exit-active"
    data-testid="element"
  >
    This is some content.
  </div>
</div>
`;

exports[`ScaleTransition should render correctly for a vertical transition 5`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;
